servicetalk-http-api/src/main/java/io/servicetalk/http/api/HttpExecutionStrategies.java

```diff
@@ -117,10 +117,10 @@ public static Builder customStrategyBuilder() {
     public static HttpExecutionStrategy difference(final Executor fallback,
                                                    final HttpExecutionStrategy left,
                                                    final HttpExecutionStrategy right) {
-        if (left.equals(right) || right == noOffloadsStrategy()) {
+        if (left.equals(right) || noOffloads(right)) {
             return null;
         }
-        if (left == noOffloadsStrategy()) {
+        if (noOffloads(left)) {
```

Comment:
We can not compare with `noOffloadsStrategy()` by reference, because we may receive a `noOffloadsStrategy()` which is wrapped by `DefaultGrpcExecutionStrategy` here or users may give their own implementation of the strategy without offloading.
@NiteshKant WDYT about introducing something like `boolean HttpExecutionStrategy.isNoOffloading()`? Or maybe even promoting this method to `ExecutionStrategy` interface. That will help us to avoid duplication of `noOffloads` method impl, we will be able to use it in `DefaultHttpExecutionStrategy.offloadService` in addition to `diff == null` check, and users will have an easy way to check that there is no offloading.

Reply:
I do not think that we will have many users checking presence/absence of no-offloads so I do not think we should promote this to a public API. The code repetition in jersey code is perhaps we can avoid by using this `difference()` method in jersey? There shouldn't be a reason for us to be different in behavior between routers. Is changing jersey to use the common `difference()` method here something you are thinking of doing?

Reply:
I see that the PR #682 changes jersey router to use its own impl of `difference` function instead of the `HttpExecutionStrategies.difference`. If I change `EndpointEnhancingRequestFilter` to use `HttpExecutionStrategies.difference`, some tests in jersey module fail.

Reply:
We can fix the tests if they are inaccurate or do not align with gRPC, there isn't a reason why they should be different. It is OK to do this in a follow up but we should look at reducing special cases between modules which makes it hard to reason about behavior later.

Reply:
Created an issue https://github.com/apple/servicetalk/issues/924